src/main/java/org/shanerx/tradeshop/player/ShopUser.java

```diff
@@ -134,6 +135,19 @@ public static Sign findObservedSign(Player observer) {
         }
     }
 
+    /**
+     * Implementation of Cantor's pairing function.
+     * Maps to integers (in this case x and z coords) to a single unique index.
+     * Ref: https://en.wikipedia.org/wiki/Pairing_function
+     *
+     * @param x the x coord
+     * @param z the z coord
+     * @return unique index of the (x,z)-tuple
+     */
+    private static int cantorPairingFunction(int x, int z) {
```

Comment:
Just make sure this isn't too intensive. Although it shouldn't be with find running aSync

Reply:
It's very simple math though, probably optimized by the JVM to run on bare metal - many times quicker than iterating through the list to figure out if the chunk is in thete